### PR TITLE
fix: handle deletion of all Docker Desktop extensions

### LINKED ISSUE #6961: Unable to delete DD extension
### Bug description

I can't remove the Disk Usage DD extension. The error in the log:

```
9:36:14 a.m. [main] Error occurred in handler for 'docker-desktop-plugin:delete': Error: Unable to find the extensionDisk usage
    at ContributionManager.deleteExtension (/Users/deboer/../podman-desktop/packages/main/dist/index.cjs:66412:13)
    at /Users/deboer/git/../podman-desktop/packages/main/dist/index.cjs:75636:41
    at WebContents.<anonymous> (node:electron/js2c/browser_init:2:82812)
    at WebContents.emit (node:events:518:28)
```

Uninstalling other DD extensions appears to have worked.

### Operating system

macOS 14.4

### Installation Method

None

### Version

next (development version)

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
There was a mismatch with some extensions with case (all lowercase vs some Upper case values)
Use the extension ID instead of the name for the deletion

another commit has been added to handle correctly the unwrapping of errors

### What issues does this PR fix or reference?

fixes #6961 

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/main/src/plugin/docker-extension/docker-desktop-installation.ts [hunk 1]
```diff
@@ -177,7 +201,7 @@ export class DockerDesktopInstallation {
       },
     );
 
-    ipcMain.handle(
+    this.ipcHandle(
       'docker-desktop-plugin:delete',
       async (event: IpcMainInvokeEvent, extensionName: string): Promise<void> => {
```
Comment (axel7083) [inline packages/main/src/plugin/docker-extension/docker-desktop-installation.ts]: Should this be renamed to extensionId ?

Comment (benoitf) [reply packages/main/src/plugin/docker-extension/docker-desktop-installation.ts]: fixed

**Review by axel7083 — APPROVED** [review]

**Review by jeffmaury — COMMENTED** [review]
Does not work for me. I installed the Disk Usage extension but can't remove it

dd

I see only parameters being renamed, am I missing something ?

**Comment by benoitf** [comment]
I think rebased failed :-)

**Comment by benoitf** [comment]
@jeffmaury please retry one commit was missing

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**